libs/services/designer-client-services/src/lib/consumption/run.ts

```diff
@@ -0,0 +1,289 @@
+  updateCors?: () => void;
+  accessToken?: string;
+  workflowId: string;
+  isDev?: boolean;
+}
+
+export class ConsumptionRunService implements IRunService {
+  _isDev = false;
+
+  constructor(public readonly options: ConsumptionRunServiceOptions) {
+    const { apiVersion, baseUrl, isDev } = options;
+    if (!baseUrl) {
+      throw new ArgumentException('baseUrl required');
+    } else if (!apiVersion) {
+      throw new ArgumentException('apiVersion required');
+    }
+    this._isDev = isDev || false;
+  }
+
+  async getContent(contentLink: ContentLink): Promise<any> {
+    const { uri, contentSize } = contentLink;
+    const { httpClient } = this.options;
+
+    // 2^18
+    if (contentSize > 262144) {
+      return undefined;
+    }
+
+    if (!uri) {
+      throw new Error();
+    }
+
+    try {
+      const response = await httpClient.get<any>({
+        uri,
+        noAuth: true,
+        headers: { 'Access-Control-Allow-Origin': '*' },
+      });
+      return response;
+    } catch (e: any) {
+      throw new Error(e.message);
+    }
+  }
+
+  private getAccessTokenHeaders = () => {
+    const { accessToken } = this.options;
+    if (!accessToken) {
+      return undefined;
+    }
+
+    return new Headers({
+      Authorization: accessToken,
+    });
+  };
+
+  async getMoreRuns(continuationToken: string): Promise<Runs> {
+    const headers = this.getAccessTokenHeaders();
+    const { httpClient } = this.options;
+
+    try {
+      const response = await httpClient.get<ArmResources<Run>>({
+        uri: continuationToken,
+        headers: headers as Record<string, any>,
+      });
+
+      const { nextLink, value: runs }: ArmResources<Run> = response;
+      return { nextLink, runs };
+    } catch (e: any) {
+      throw new Error(e.message);
+    }
+  }
+
+  /**
+   * Gets run details.
+   * @param {string} runId - Run id.
+   * @returns {Promise<Run>} Workflow runs.
+   */
+  async getRun(runId: string): Promise<Run> {
+    const { apiVersion, baseUrl, httpClient } = this.options;
+
+    const uri = `${baseUrl}${runId}?api-version=${apiVersion}&$expand=properties/actions,workflow/properties`;
+
+    try {
+      const response = await httpClient.get<Run>({
+        uri,
+      });
+      return response;
+    } catch (e: any) {
+      throw new Error(e.message);
+    }
+  }
+
+  /**
+   * Gets workflow run history
+   * @returns {Promise<Runs>} Workflow runs.
+   */
+  async getRuns(): Promise<Runs> {
+    const { apiVersion, baseUrl, workflowId, httpClient } = this.options;
+    const headers = this.getAccessTokenHeaders();
+
+    const uri = `${baseUrl}${workflowId}/runs?api-version=${apiVersion}`;
+    try {
+      const response = await httpClient.get<ArmResources<Run>>({
+        uri,
+        headers: headers as Record<string, any>,
+      });
+
+      const { nextLink, value: runs }: ArmResources<Run> = response;
+      return { nextLink, runs };
+    } catch (e: any) {
+      throw new Error(e.message);
```

Comment:
Is there a reason why we try catch if we just throw another Error with the same message?